src/firefly/java/edu/caltech/ipac/firefly/server/visualize/WebPlotReader.java

```diff
@@ -401,5 +406,48 @@ private static double getImageSize(WebPlotRequest r) {
         }
     }
 
+    private static List<List<RelatedData>> investigateRelations(File f, FitsRead frAry[]) {
+        if (frAry.length>1) {
+            int imageIdx= -1;
+            List<List<RelatedData>> retList= new ArrayList<>(frAry.length);
+            for(int i=0; (i<frAry.length); i++) {
+                retList.add(null);
+                String extType= frAry[i].getExtType();
+                if (extType!=null && extType.equalsIgnoreCase("IMAGE")) {
+                    imageIdx = i;
+                }
+            }
+            if (imageIdx>-1) {
+                List <RelatedData> relatedList= new ArrayList<>();
+                for(int i=0; (i<frAry.length); i++) {
+                    String extType= frAry[i].getExtType();
+                    if (extType!=null) {
+                        if (!extType.equalsIgnoreCase("IMAGE")) {
+                            if (frAry[i].getExtType().equalsIgnoreCase("MASK")) {
```

Comment:
can you combine these two if statements into one: 
if (extType.equalsIgnoreCase("MASK")?